### PR TITLE
feat(telemetry): report host_id for atlas COMPASS-8092

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

For every detected Atlas cluster track the first resolved hostname without the port when connecting.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/telemetry/connectionTelemetry.ts [hunk 1]
```diff
@@ -56,11 +57,8 @@ async function getHostnameForConnection(
 }
 
 async function getCloudInfoFromDataService(
```
Comment (mcasimir) [inline src/telemetry/connectionTelemetry.ts:59]: should this fn be renamed to `getCloudInfoFromHostname` now?

File: src/telemetry/connectionTelemetry.ts [hunk 2]
```diff
@@ -110,15 +108,22 @@ export async function getConnectionTelemetryProperties(
     const username = connectionString.username ? 'DEFAULT' : 'NONE';
     const authStrategy = authMechanism ?? username;
 
+    const hostname = await getHostnameForConnection(
+      dataService.getConnectionString()
```
Comment (mcasimir) [inline src/telemetry/connectionTelemetry.ts:112]: Looks like you have `const connectionString = dataService.getConnectionString();` on line `106` already available

File: src/telemetry/connectionTelemetry.ts [hunk 3]
```diff
@@ -110,15 +108,22 @@ export async function getConnectionTelemetryProperties(
...
+      getCloudInfoFromDataService(hostname),
     ]);
+    const isAtlas = mongoDBBuildInfo.isAtlas(connectionString.toString());
+    const atlasHostId = isAtlas ? hostname : null;
 
     preparedProperties = {
       ...preparedProperties,
       auth_strategy: authStrategy,
-      is_atlas: mongoDBBuildInfo.isAtlas(connectionString.toString()),
+      is_atlas: isAtlas,
```
Comment (mcasimir) [inline src/telemetry/connectionTelemetry.ts:125]: wow interesting that we didn't have this one :)

**Review by mcasimir — APPROVED** [review]
LGTM, left a couple of NIT comments. Couple of thoughts that came to mind while reviewing:
- can we add a test to make sure that we don't send this host id for non atlas connections?
- I liked the idea of calling this prop `atlas_host_id` as you did for the constant, instead of `host_id`, we can quickly ask in the analytics channel if they would be ok with it?

File: src/telemetry/connectionTelemetry.ts (hunk 3 above)
Comment (alenakhineika) [reply src/telemetry/connectionTelemetry.ts:125]: We did have. I just read it from `mongoDBBuildInfo.isAtlas()` higher in code now.

Comment (mcasimir) [reply src/telemetry/connectionTelemetry.ts:120]: right! 🤦

**Pull request merged**